subprojects/platform-play/src/main/java/org/gradle/play/internal/twirl/TwirlCompilerAdapterV10X.java

```diff
@@ -76,6 +76,11 @@ public ScalaMethod getCompileMethod(ClassLoader cl) throws ClassNotFoundExceptio
         };
     }
 
+    @Override
+    public Object[] createCompileParameters(ClassLoader cl, File file, File sourceDirectory, File destinationDirectory, TwirlImports defaultImports, TwirlTemplateFormat templateFormat, List<String> additionalImports, List<String> constructorAnnotations) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
```

Comment:
I think we should just change the existing `createCompileParameters` and assert that constructorAnnotations are empty for older Twirl versions.  Otherwise, the TwirlCompiler change above won't call the right method.

Reply:
Agree